shapeshift/lib
# fix: osmo parser fixes

Parse osmo & cosmos tx history correctly for swaps.

To test:

make sure you are using the best cosmos/osmosis nodes, see: https://github.com/shapeshift/web/pull/2375

link locally, swap `atom -> osmo` and `osmo -> atom`. see transactions in history are for correct amounts of outgoing and incoming tokens.

NOTE ABOUT A KNOWN ISSUE:

You will not see incoming "atom" after an osmo->atom trade come over the websockets. you will only see this tx after reloading the page. This is a known issue: https://github.com/shapeshift/lib/issues/961

Screenshot below demonstrates  `atom -> osmo` followed by `osmo -> atom` (read from the bottom up)

You will notice transactions with no amounts, these are the `ibc atom on osmo` tokens being sent and received ephemerally during the trade

<img width="376" alt="Screen Shot 2022-08-05 at 8 44 18 AM" src="https://user-images.githubusercontent.com/6187559/183103657-e33da257-9243-44d1-b8b9-4002c53bdb39.png">
